src/Symfony/Component/Finder/Iterator/MultiplePcreFilterIterator.php

```diff
@@ -87,7 +85,20 @@ protected function isAccepted($string)
      */
     protected function isRegex($str)
     {
-        return Expression::create($str)->isRegex();
+        if (preg_match('/^(.{3,}?)[imsxuADU]*$/', $str, $m)) {
+            $start = substr($m[1], 0, 1);
+            $end = substr($m[1], -1);
+
+            if ($start === $end) {
+                return !preg_match('/[*?[:alnum:] \\\\]/', $start);
+            }
+
+            if ($start === '{' && $end === '}') {
```

Comment:
btw, the logic in the Expression subsystem was supporting ``()`` as delimiters, so this is actually a regression